doc/adr/0005-tsp-update-experiment.md

```diff
@@ -0,0 +1,58 @@
+# 5. Tsp update experiment
+
+Date: 2022-03-22
+
+## Status
+
+Proposed
+
+## Context
+
+The tentative [TSP specification][tsp] proposes a PUT command on the `tsp/experiments/{expUUID}` endpoint for the following use cases:
+
+- Rename experiment
+- Add/remove traces to/from an experiment
+- Other updates (e.g. offsetting of traces)
+
+When an experiment is created the client passes the `name` of the experiment and a `list of trace UUIDs`  to the POST request.
+It returns a data structure with experiment details and an `unique UUID`, to identify the experiment.
+
+This ADR proposes the way-forward for renaming and adding/removing of traces of experiments. It doesn't cover other updates like offsetting of traces. They will be handled in other ADRs.
+
+PUT commands require to be idempotent, multiple requests have to have the same outcome. For a PUT to be idempotent, it is required to provide all parameters that makes up an experiment to be sent in the PUT command. In the case of updating of an experiment, the `name` and `list of trace UUIDs` have to be present, even if some parts don't change. The PUT command replaces the experiment resource on the server with a new one.
+
+PATCH commands don't have that requirement of having to be idempotent. They can be idempotent, but don't have to. PATCH can be used to update parts of the experiment resource. For example, when updating the name of an experiment only the `name` has to be passed in the PATCH request, and not the `list of trace UUIDs`.
+
+For both PUT and PATCH, the experiment UUID must not change. The Trace Server implementation needs to be able to provide that.
+
+For updating of an experiment PATCH would be more suitable because only the name to be modified needs to be provided.
+
+`Notes`:
+
+- The current Trace Compass server implementation, creates the UUID based on the experiment name and persists the experiment as folder resource in the server workspace. Renaming an experiment and renaming the corresponding workspace resource would lead to a change of the UUID as well. Hence, the server implemenation needs to be changed to return the same UUID.
```

Comment:
Extra comma in `implementation, creates` it seems.

Reply:
Done